Book-A-Desk.Infrastructure/ReservationInstructionInterpreter.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Book_A_Desk.Infrastructure
+
+open Book_A_Desk.Domain.Events
+open Book_A_Desk.Domain.Office.Domain
+open Book_A_Desk.Domain.Reservation
+open Book_A_Desk.Domain.Reservation.Commands.Instructions.ReservationInstructions
+open Book_A_Desk.Domain.Reservation.Domain
+
+module ReservationInstructionInterpreter =
+    
+    // Note: To make the program run, all you need to do is:
+    // ReservationsCommandHandlerUsingInstructions.handle command
+    // |> ReservationInstructionInterpreter.interpret
+    
+    let rec interpret (eventStore : EventStore) program = async {
```

Comment:
Here's the interpreter, who recursively calls itself with the different commands. This means that the program is only really executed when we call the interpreter and not when someone calls `ReservationsCommandHandlerUsingInstructions.handle`.

As you can see, the code is asynchronous and that is completely transparent to the domain.